src/Framework/FieldsAPI/FormField/FieldOptions.php

```diff
@@ -2,29 +2,90 @@
 
 namespace Give\Framework\FieldsAPI\FormField;
 
-use Give\Framework\FieldsAPI\FormField\FieldTypes;
+use Give\Helpers\Arr;
 
 trait FieldOptions {
 
-	/** @var array */
+	/** @var FieldOption[] */
 	protected $options = [];
 
+	/**
+	 * Does the field type support options?
+	 *
+	 * @return bool
+	 */
 	public function supportsOptions() {
 		return in_array(
 			$this->getType(),
 			[
 				FieldTypes::TYPE_SELECT,
 				FieldTypes::TYPE_RADIO,
-			]
+			],
+			true
 		);
 	}
 
-	public function options( $options ) {
-		$this->options = $options;
+	/**
+	 * Set the options
+	 *
+	 * @param array $options
+	 *
+	 * @return $this
+	 */
+	public function options( array $options ) {
+		// Reset options, since they are meant to be set immutably
+		$this->options = [];
+
+		// Determine if the provided options are an associative array before the loop to avoid extra work.
+		$areOptionsAnAssocArray = Arr::isAssoc( $options );
+
+		// Loop through the options and transform them to the proper format.
+		foreach ( $options as $maybeLabel => $fieldOptionOrArrayOrValue ) {
```

Comment:
@knowler I'm implementing your changes into P2P, great work btw, but I have one question. Shouldn't the current option key be the option value instead of label? I'm asking because I will have to swap the [value with label](https://github.com/impress-org/give-peer-to-peer/blob/5bc87e8bd8fa3c53e34410fd53e321f17084c647/src/P2P/Admin/Settings/CampaignSettings.php#L80) in order to display the value and label correctly on P2P.

Reply:
Nice catch. Yep, that's how it should be and even how I wrote the example. I'll fix that here.

Reply:
Awesome, thanks!

Reply:
Resolved in 8dd3b29